metagraph/src/common/utils/string_utils.cpp

```diff
@@ -15,6 +15,11 @@ bool ends_with(const std::string &str, const std::string &suffix) {
     return actual_suffix == suffix;
 }
 
+bool starts_with(const std::string &str, const std::string &prefix) {
+    auto actual_prefix = str.substr(0, static_cast<int>(prefix.size()));
+    return actual_prefix == prefix;
```

Comment:
1. check length of the string (`prefix` might be longer than `str`)
2. use string_buffer instead of actually creating a new string `actual_prefix`

Reply:
i missed those cases. Did the changes, what do you think?